diff --git a/index.theme b/index.theme
index a22607c6..d8078c8d 100644
--- a/index.theme
+++ b/index.theme
@@ -1,5 +1,5 @@
 [Icon Theme]
-Name=La Capitaine
+Name=La Capitaine (Dark)
 Comment=La Capitaine is an icon pack inspired by macOS and Google Material Design - a beautiful hybrid between the two most popular design specifications to date.
 Inherits=breeze,elementary,gnome,deepin
 Directories=actions/symbolic,actions/22x22,animations/22x22,apps/scalable,apps/symbolic,devices/symbolic,devices/scalable,emblems/scalable,emblems/symbolic,emotes/scalable,mimetypes/scalable,mimetypes/symbolic,places/16x16,places/scalable,places/symbolic,panel/16,panel/24,status/scalable,status/symbolic
